### PR TITLE
Drop Python 3.5 support

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description

xref: https://github.com/numpy/numpy/pull/14086
closes https://github.com/scikit-image/scikit-image/issues/4100

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by hmaarrfk** [comment]
3.8 is too hard to add.

**Comment by hmaarrfk** [comment]
Let me know how you want it to be changed.

**Comment by sciunto** [comment]
Indeed, it's more than a guarantee, 0.16 won't be compatible as we will use f-string in many places very soon.

**Comment by sciunto** [comment]
@hmaarrfk Do you want to propose a solution to the conflict?

**Comment by soupault** [comment]
Are we going to make the releases for Py3.5 still (https://github.com/scikit-image/scikit-image/blob/master/setup.py#L218)? 🤔

**Comment by hmaarrfk** [comment]
No, no releases for 3.5. Thanks for the catch. Hopefully I updated the setup.py correctly.

File: setup.py [hunk 1]
```diff
@@ -36,11 +36,11 @@
 from numpy.distutils.command.build_ext import build_ext
 
 
-if sys.version_info < (3, 5):
+if sys.version_info < (3, 6):
 
     error = """Python {py} detected.
 
-scikit-image 0.15+ support only Python 3.5 and above.
+scikit-image 0.16+ support only Python 3.6 and above.
```
Comment (jni) [inline setup.py]:
```suggestion
scikit-image 0.16+ supports only Python 3.6 and above.
```

File: doc/release/release_dev.rst [hunk 2]
```diff
@@ -12,6 +12,8 @@ For more information, examples, and documentation, please visit our website:
 https://scikit-image.org
 
 
+This release of scikit-image drops support for Python 3.5.
+This release of scikit-image officially supports Python 3.6 and 3.7.
```
Comment (jni) [inline doc/release/release_dev.rst]: Perhaps we should state the policy going forward?

Comment (hmaarrfk) [reply doc/release/release_dev.rst]: I thought release notes were about facts, not decision process. Maybe it can go somewhere else?

Comment (jni) [reply doc/release/release_dev.rst]: When the API or supported versions change, affected users typically want to know why. It's the reason why we "link" to the PRs, I think. You could add a "see #4102 for more info", but you can explain the NEP in about the same space.

Comment (soupault) [reply doc/release/release_dev.rst]: How about something like this? :
"Starting from this release, scikit-image will follow the spirit of the recently introduced numpy deprecation policy -- NEP 29 (https://github.com/numpy/numpy/blob/master/doc/neps/nep-0029-deprecation_policy.rst). Accordingly, scikit-image 0.16 drops the support for Python 3.5 [and numpy __]. (#4102)"

**Comment by soupault** [comment]
Thanks, Mark!

**Pull request merged**